### PR TITLE
TMA-762: Fix test for swapping date dimensions

### PR DESCRIPTION
**Pull request by panjan**

### CODE REVIEW

File: spec/integration/date_dim_switch_spec.rb [hunk 1]
```diff
@@ -135,10 +139,10 @@ def suggest_mapping(label_a, label_b, project)
     expect(@report.definition.attributes.map(&:identifier)).to eq ["created_on_2.quarter"]
 
     ids = GoodData::SmallGoodZilla.get_uris(@report.definition.filters.first).map { |x| x.split('/')[-2..-1].join('/') }
-    expect(ids).to eq ["obj/281", "281/elements?id=2015", "281/elements?id=2016"]
+    expect(ids).to eq ["obj/#{year_obj_id}", "#{year_obj_id}/elements?id=2015", "#{year_obj_id}/elements?id=2016"]
```
Comment (yenkins) [inline spec/integration/date_dim_switch_spec.rb]: Identical code found in date_dim_switch_spec.rb:142, date_dim_switch_spec.rb:145 (mass = 64)

File: spec/integration/date_dim_switch_spec.rb [hunk 2]
```diff
@@ -135,10 +139,10 @@ def suggest_mapping(label_a, label_b, project)
...
 
     ids = GoodData::SmallGoodZilla.get_uris(@variable.values.first.expression).map { |v| v.split('/')[-2..-1].join('/') }
-    expect(ids).to eq ["obj/281", "281/elements?id=2015", "281/elements?id=2016"]
+    expect(ids).to eq ["obj/#{year_obj_id}", "#{year_obj_id}/elements?id=2015", "#{year_obj_id}/elements?id=2016"]
```
Comment (yenkins) [inline spec/integration/date_dim_switch_spec.rb]: Identical code found in date_dim_switch_spec.rb:142, date_dim_switch_spec.rb:145 (mass = 64)

File: spec/integration/date_dim_switch_spec.rb [hunk 3]
```diff
@@ -110,6 +110,10 @@ def suggest_mapping(label_a, label_b, project)
     expect(as.pmapcat { |a| a.usedby('metric') }).to be_empty
     expect(as.pmapcat { |a| a.usedby('dashboard') }).to be_empty
 
+    year_obj_id = @project.attributes
+                          .find{ |a| a.identifier == 'created_on_2.year'}
```
Comment (yenkins) [inline spec/integration/date_dim_switch_spec.rb]: Style/SpaceBeforeBlockBraces: Space missing to the left of {.

Comment (yenkins) [inline spec/integration/date_dim_switch_spec.rb]: Style/SpaceInsideBlockBraces: Space missing inside }.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/652/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/443/ : FAILURE in 2m 16s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/751/ : SUCCESS in 2m 08s

File: spec/integration/date_dim_switch_spec.rb (hunk 2 above)
Comment (kubamahnert) [inline spec/integration/date_dim_switch_spec.rb]: why do we still reference objects by hard coded ids? `elements?id=2016`

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/653/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/444/ : FAILURE in 33s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/752/ : SUCCESS in 1m 52s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/654/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/445/ : FAILURE in 32s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/753/ : SUCCESS in 2m 05s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/655/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/446/ : SUCCESS in 1m 43s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/754/ : SUCCESS in 40s

**Comment by kubamahnert** [comment]
ok to merge

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/249/ : SUCCESS in 14s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/447/ : SUCCESS in 30s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/755/ : SUCCESS in 32s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1165/ : SUCCESS in 49m 27s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/923/ : SUCCESS in 32m 28s

**Pull request merged**